### ISSUE TITLE
`.remove(-1)` doesn't work.

### ISSUE THREAD

**Issue by bend-n** [godotengine/godot-proposals#4637]
### Describe the project you are working on

Chess, but can apply to many projects.

### Describe the problem or limitation you are having in your project

Can not use `.remove(-1)` on Arrays.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Looks clean, and it doesn't make sense that you cant use (-1) indexing.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

`[1, 2].remove(-1)`

### If this enhancement will not be used often, can it be worked around with a few lines of script?

`[1, 2].remove([1, 2].size() -1)`, yes.
However, its rather unclean.

### Is there a reason why this should be core and not an add-on in the asset library?

You cant put extra functionality in array; It would have to take a array, and return a array, which is not particularly clean.

**Comment by KoBeWi** [comment]
Why not just `pop_back()`?

**Comment by YuriSizov** [comment]
I agree with KoBeWi that for -1 specifically we already have a method that works. If your intention is for general negative indices to be supported, that may be worth discussing. We do have some methods and accessors that support negative indices, so it wouldn't be that out of place.

**Comment by bend-n** [comment]
> Why not just `pop_back()`?

`pop_back()` does not exist on Pool*Arrays

**Comment by bend-n** [comment]
We should just add `pop_back` for `Packed*Array`

**Comment by AThousandShips** [comment]
For `array` you can just use `pop_at` for `remove(-x)`

**Comment by bend-n** [comment]
huh how did this get through
